include/cmetrics/cmt_compat.h

```diff
@@ -25,6 +25,28 @@
 #include <windows.h>
 #endif
 
+/* This function is copied from monkey/monkey.
+   https://github.com/monkey/monkey/blob/2567a70912ed7a68d9e75dca3cf22d3927fea99a/mk_core/deps/libevent/evdns.c#L3323 */
+static inline char *
+cmt_platform_strtok_r(char *s, const char *delim, char **state) {
```

Comment:
it seems this function is also modifying the source string which we want to avoid. we should not modify the original buffer.

Reply:
I tried to preserve the original string. Then, I realized that this shouldn't be feasible only three arguments.
I added the fourth one which handles allocated memory. This introduced complexity could mitigate the requirements:
https://github.com/fluent/cmetrics/pull/210/commits/268543872a08d2f36afeae3501cc6534a050f127